yoti-sdk-api/src/main/java/com/yoti/api/client/ActivityDetails.java

```diff
@@ -13,26 +12,26 @@ public final class ActivityDetails {
 
     private final String rememberMeId;
     private final String parentRememberMeId;
-    private final ApplicationProfile applicationProfile;
-    private final HumanProfile userProfile;
+    private final ApplicationProfile thisPartyProfile;
+    private final HumanProfile otherPartyProfile;
     private final Date timestamp;
     private final String receiptId;
     private final ExtraData extraData;
 
     public ActivityDetails(String rememberMeId,
             String parentRememberMeId,
-            HumanProfile userProfile,
-            ApplicationProfile applicationProfile,
+            HumanProfile otherPartyProfile,
+            ApplicationProfile thisPartyProfile,
             ExtraData extraData,
             Date timestamp,
             byte[] receiptId) {
-        this.rememberMeId = notNull(rememberMeId, "Remember Me id");
+        this.rememberMeId = rememberMeId;
         this.parentRememberMeId = parentRememberMeId;
-        this.userProfile = notNull(userProfile, "User profile");
-        this.applicationProfile = notNull(applicationProfile, "Application profile");
+        this.otherPartyProfile = otherPartyProfile;
+        this.thisPartyProfile = thisPartyProfile;
```

Comment:
I could be missing something, but `ActivityDetailsFactory` is always passing non null instances of `HumanProfile` and `ApplicationDataProfile`:
```java
public ActivityDetails create(Receipt receipt, PrivateKey privateKey) throws ProfileException {
    (...)
    
    HumanProfile otherPartyProfile = new HumanProfile(otherPartyAttr);
    ApplicationProfile thisPartyProfile = new ApplicationProfile(thisPartyAttr);
    
    (...)
    
    return new ActivityDetails(
            (...)
            otherPartyProfile,
            thisPartyProfile,
           (...)
    );
}
```

So while the _content_ of those instances might be `null` (though I would expect them to be empty lists instead of `null` but I didn't check the codebase to be sure) the instances themselves (of `HumanProfile` and `ApplicationDataProfile`) are never `null`.

Whether or not the checks should be kept, that's another question, but personally if it's being done for other fields why not these if they are indeed never `null`?

Reply:
I triple checked and these should never be `null` (if not for a bug) I'll put the check back. Good spot.